ee/tables/windowsupdatetable/windowsupdate_cached.go

```diff
@@ -0,0 +1,116 @@
+//go:build windows
+// +build windows
+
+package windowsupdatetable
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/dataflatten"
+	"github.com/kolide/launcher/ee/observability"
+	"github.com/kolide/launcher/ee/tables/dataflattentable"
+	"github.com/kolide/launcher/ee/tables/tablehelpers"
+	"github.com/kolide/launcher/ee/tables/tablewrapper"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+const (
+	cachedDataTTL = 6 * time.Hour
+)
+
+type CachedWindowsUpdatesTable struct {
+	slogger    *slog.Logger
+	cacheStore types.Getter
+	name       string
+}
+
+func CachedWindowsUpdatesTablePlugin(flags types.Flags, slogger *slog.Logger, cacheStore types.Getter) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("locale"),
+		table.IntegerColumn("is_default"),
+	)
+
+	t := &CachedWindowsUpdatesTable{
+		slogger:    slogger.With("name", "kolide_windows_updates_cached"),
+		cacheStore: cacheStore,
+		name:       "kolide_windows_updates_cached",
```

Comment:
I still feel a bit biased away from a different table name.

Reply:
Do you mean you want to name this table something else, or that you prefer to keep this inside `kolide_windows_updates` with a `cached` column? If the former, happy to rename -- let me know what you don't like about this one or what you'd prefer. If the latter -- I chose a separate table to make it easier for us to rewrite SQL on the fly, if that's the route we end up taking. (Rewriting the SQL to add/remove a column constraint sounds painful, but replacing `kolide_windows_updates_cached` with `kolide_windows_updates` or vice versa should be easier.)

Reply:
I had been thinking about column constraints, but that's a compelling arguement in how we handle re-writing it.